cupy/cuda/compiler.py

```diff
@@ -138,7 +138,7 @@ def compile_with_cache(source, options=(), arch=None, cache_dir=None):
     mod = function.Module()
 
     if not os.path.exists(cache_dir):
-        os.makedirs(cache_dir)
+        os.makedirs(cache_dir, exist_ok=True)
```

Comment:
This argument is only supported python3.
https://docs.python.jp/3/library/os.html#os.makedirs